how-to/integrate-with-sso/integrate-with-okta/client/src/auth.ts

```diff
@@ -0,0 +1,71 @@
+import { getSettings } from "./settings";
+import type { OktaSettings } from "./shapes";
+
+let authSettings: OktaSettings;
+let informationCallback: (info: string) => void;
+
+export async function init(settings: OktaSettings, informationCb: (info: string) => void) {
+	informationCallback = informationCb;
+
+	informationCallback("Initialising the authentication");
+
+	authSettings = settings;
+	if (!authSettings) {
+		informationCallback("Error: Settings missing cannot continue");
+	}
+}
+
+function makeid(length): string {
```

Comment:
we have a uuid.ts file that you can see in the auth0 src folder which you can copy. This gives you a secure uuid if possible using browser apis or an alternative if it is http and not localhost.

Reply:
done